plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -368,6 +369,22 @@ protected void verifyColumnName(DatabaseMetaData databaseMetadata, String column
         }
     }
 
+    @Override
+    public OptionalInt getMaxColumnNameLength(ConnectorSession session)
+    {
+        try (Connection connection = connectionFactory.openConnection(session)) {
```

Comment:
Looks like this is copied/pasted in a few places?

Reply:
Yes. I can probably move this into trino-base-jdbc as a default impl. to be shared for drivers which have properly implemented getMaxColumnNameLength.

Reply:
I'm not a big fan of sharing code via inheritance. To me it makes more sense to close that common code in a separate class and inject the instance via DI (possibly swapping for something else via modules or annotation mechanism). For me the inheritance-way never worked long term. Not sure what patterns/conclusions WRT inheritance/no-inheritance exist in this project.

Reply:
> I'm not a big fan of sharing code via inheritance.

@dominikzalewski Me too, but (un)fortunately that was a design decision once framework for JDBC connector was created. That the only thing one should should be extending `JdbcClient` (and use `BaseJdbcClient` for convenience) and that should be enough to implement. All SPI and Trino things should be covered by `trino-base-jdbc` framework.

https://trino.io/docs/current/develop/example-jdbc.html

Reply:
extracted as a helper to BaseJdbClient, each connector still needs to opt-in to this.

See `TopNFunction` and `limitFunction` in `BaseJdbcClient` as the pattern which was followed.